include/deal.II/particles/generators.h

```diff
@@ -162,6 +164,84 @@ namespace Particles
+     * The total number of particles that is added to the @p particle_handler object is
+     * the number of dofs of the DoFHandler that is passed that are within the
+     * triangulation and whose components are within the ComponentMask.
+     *
+     * @param[in] a DOF handler that may live on another triangulation that is
+     * used to establsh the positions of the particles
+     *
+     * @param[in] A vector that contains all the bounding boxes for all
+     * processor. This vector can be established by first using
+     * 'GridTools::compute_mesh_predicate_bounding_box()' and gathering all the
+     * bounding boxes using 'Utilities::MPI::all_gather.
+     *
+     * @param[in,out] particle_handler The particle handler that will take
+     * ownership of the generated particles.
+     *
+     * @param[in] mapping An optional mapping object that is used to map
+     * the DOF locations. If no mapping is provided a MappingQ1 is assumed.
+     *
+     * @param[in] Component mask of the dof_handler from which this originates
+     *
+     * @author Bruno Blais, Luca Heltai, 2019
+     *
+     *
+     */
+    template <int dim, int spacedim = dim>
+    void
+    non_matching_dof_support_points(
+      const DoFHandler<dim, spacedim> &particle_dof_handler,
+      const std::vector<std::vector<BoundingBox<spacedim>>>
+        &                             global_bounding_boxes,
+      ParticleHandler<dim, spacedim> &particle_handler,
+      const Mapping<dim, spacedim> &  mapping =
+        StaticMappingQ1<dim, spacedim>::mapping,
+      const ComponentMask &components = ComponentMask());
+
+    /**
+     * A function that generates particles at the location of the quadrature
+     * point of a non-matching triangulation.
+     * The total number of particles that is added to the @p particle_handler object is the
+     * number of cells multiplied by the number of particle reference locations
+     * which are generally constructed using a quadrature
+     *
+     * @param[in] triangulation The non-matching triangulation which is used to
+     * insert the particles into the domain.
+     *
+     * @param[in] particle_reference_location The particle reference location
+     * that correspond to where the particles will be inserted within the cells
+     *
+     * @param[in] A vector that contains all the bounding boxes for all
+     * processor. This vector can be established by first using
+     * 'GridTools::compute_mesh_predicate_bounding_box()' and gathering all the
+     * bounding boxes using 'Utilities::MPI::all_gather.
+     *
+     * @param[in,out] particle_handler The particle handler that will take
+     * ownership of the generated particles.
+     *
+     * @param[in] mapping An optional mapping object that is used to map
+     * the DOF locations. If no mapping is provided a MappingQ1 is assumed.
+     *
+     * @author Bruno Blais, Luca Heltai, 2019
+     *
+     *
+     */
```

Comment:
no need for the empty lines at the end of the comment; the same applies to the other function